pybamm/models/submodels/electrolyte_current.py

```diff
@@ -0,0 +1,118 @@
+        eta_0p = leading_order_model.variables[
+            "Positive electrode overpotential"
+        ].orphans[0]
+        Phi_0 = leading_order_model.variables["Electrolyte potential"].orphans[0]
+        V_0 = leading_order_model.variables["Voltage"].orphans[0]
+
+        # Independent variables
+        x_n = pybamm.SpatialVariable("x", ["negative electrode"])
+        x_s = pybamm.SpatialVariable("x", ["separator"])
+        x_p = pybamm.SpatialVariable("x", ["positive electrode"])
+
+        # First-order concentration (c = c_e_0 + C_e * c_e_1)
+        c_e_1 = (c_e - c_e_0) / param.C_e
+        c_e_n = c_e.orphans[0]
+        c_e_1n = (c_e_n - c_e_0) / param.C_e
+        c_e_p = c_e.orphans[2]
+        c_e_1p = (c_e_p - c_e_0) / param.C_e
+
+        # Pre-define functions of leading-order variables
+        chi_0 = param.chi(c_e_0)
+        kappa_0n = param.kappa_e(c_e_0) * eps_0n ** param.b
+        kappa_0s = param.kappa_e(c_e_0) * eps_0s ** param.b
+        kappa_0p = param.kappa_e(c_e_0) * eps_0p ** param.b
+        j0_0n = pybamm.interface.exchange_current_density(
+            c_e_0, domain=["negative electrode"]
+        )
+        j0_0p = pybamm.interface.exchange_current_density(
+            c_e_0, domain=["positive electrode"]
+        )
+        U_0n = param.U_n(c_e_0)
+        U_0p = param.U_p(c_e_0)
+        j0_1n = c_e_1n * j0_0n.diff(c_e_0)
+        j0_1p = c_e_1p * j0_0p.diff(c_e_0)
+        dU_0n__dc0 = U_0n.diff(c_e_0)
+        dU_0p__dc0 = U_0p.diff(c_e_0)
+
+        # Potential
+        cbar_1n = pybamm.Integral(c_e_1n, x_n) / param.l_n
+        j0bar_1n = pybamm.Integral(j0_1n, x_n) / param.l_n
+        A_n = (
+            j0bar_1n * pybamm.Function(np.tanh, eta_0n) / j0_0n
+            - dU_0n__dc0 * cbar_1n
+            - chi_0 / c_e_0 * cbar_1n
+            + i_cell * param.l_n / (6 * kappa_0n)
+        )
+
+        Phi_1n = -i_cell * x_n ** 2 / (2 * param.l_n * kappa_0n)
+        Phi_1s = -i_cell * ((x_s - param.l_n) / kappa_0s + param.l_n / (2 * kappa_0n))
+        Phi_1p = -i_cell * (
+            param.l_n / (2 * kappa_0n)
+            + param.l_s / (kappa_0s)
+            + (param.l_p ** 2 - (1 - x_p) ** 2) / (2 * param.l_p * kappa_0p)
+        )
+        Phi_1 = (
+            chi_0 / c_e_0 * c_e_1
+            + pybamm.Concatenation(
+                pybamm.Broadcast(Phi_1n, ["negative electrode"]),
+                pybamm.Broadcast(Phi_1s, ["separator"]),
+                pybamm.Broadcast(Phi_1p, ["positive electrode"]),
+            )
+            + A_n
+        )
+
+        # Voltage
+        cbar_1p = pybamm.Integral(c_e_1p, x_p) / param.l_p
+        Phibar_1p = pybamm.Integral(Phi_1p, x_p) / param.l_p
+        j0bar_1p = pybamm.Integral(j0_1p, x_p) / param.l_p
+        V_1 = (
+            Phibar_1p
+            + dU_0p__dc0 * cbar_1p
+            - j0bar_1p * pybamm.Function(np.tanh, eta_0p) / j0_0p
+        )
+
+        # Variables
+        self.variables = {
+            "c": c_e,
```

Comment:
Should these outputs be "c_e" and "phi_e" so as not to name clash elsewhere?

Reply:
Also should we output the current and the flux aswell since other submodels do aswell?

Reply:
Also should we be only outputting the potentials here and not "c_e"